src/java.base/share/classes/java/io/FileDescriptor.java

```diff
@@ -61,21 +61,63 @@ public final class FileDescriptor {
+
+                OpenResourcePolicies.Policy policy = findPolicy(nativeDescription);
+                String action = "error";
+                String warn = "false";
+                Supplier<Exception> supplier = null;
+                // Normally the claiming should be overridden by FileInputStream/FileOutputStream
+                // but in case these are collected we handle FDs 0..2 here as well.
+                if (policy != null) {
+                    action = policy.action;
+                    warn = policy.params.getOrDefault("warn", "true");
+                } else if (self == in || self == out || self == err) {
+                    action = "ignore";
+                }
+                supplier = switch (action.toLowerCase()) {
+                    case "error":
+                        yield () -> new CheckpointOpenResourceException(
+                            FileDescriptor.class.getSimpleName() + " " + fd + ": " + nativeDescription,
+                            getStackTraceHolder());
+                    case "close":
+                        close();
+                    case "ignore":
+                        if (Boolean.parseBoolean(warn)) {
+                            LoggerContainer.warn("File descriptor {0} was not closed by the application. Use 'warn: false' in the policy to suppress this message.", fd);
+                        }
+                        yield NO_EXCEPTION;
+                    default: throw new IllegalArgumentException("Unknown policy action for file descriptor " + fd + ": " + action);
+                };
+                claimedFDs.claimFd(self, self, supplier);
+            }
+        }
 
-                claimedFDs.claimFd(self, self, () ->  {
-                    if (self == in || self == out || self == err) {
-                        return null;
+        private OpenResourcePolicies.Policy findPolicy(String nativeDescription) {
+            return OpenResourcePolicies.find(false, "filedescriptor", params -> {
+                String value = params.get("value");
+                if (value != null) {
+                    try {
+                        int expected = Integer.parseInt(value);
+                        if (expected != fd) {
+                            return false;
+                        }
+                    } catch (NumberFormatException e) {
+                        throw new IllegalArgumentException("Cannot parse file descriptor value '" + value + "'");
                     }
-                    return new CheckpointOpenResourceException(
-                        FileDescriptor.class.getSimpleName() + " " + fd + ": " + nativeDescription0(),
-                        getStackTraceHolder());
-                });
-            }
+                }
+                String regex = params.get("regex");
+                if (regex != null) {
+                    return Pattern.compile(regex).matcher(nativeDescription).matches();
```

Comment:
I think `.find()` instead `.matches()` will fit better. The matches() looks for the complete match, while find() will allow substring match specification. Complete match specification will still be possible with `^` and/or `$`.